### PR TITLE
Add CI pipeline to execute tests

### PR DESCRIPTION
**Pull request by luca-byte**
Adds a pipeline, using the official ros actions, to test the environment

- Add the workflow
- Add some ros lint tests
- Fix the codebase so that tests pass
- Add VSCode configuration instructions in `README.md` for the formatter

I am conscious that this PR is disgusting (807 additions 🥲) but it should have mostly style changes. However you should check it, since I had to make some changes also to the code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,9 +16,69 @@ git clone --recursive https://github.com/Team-Isaac-Polito/reseq_ros2.git src
...
+
+```
+colcon test --event-handlers=console_direct+
+```
+
+## Gazebo Simulator
+Due to problems with wsl, the project uses _Gazebo Harmonic_, which can be installed from the official Gazebo repositories.
+
+```bash
+sudo curl https://packages.osrfoundation.org/gazebo.gpg --output /usr/share/keyrings/pkgs-osrf-archive-keyring.gpg
```
Comment (Yunus-Selim) [inline README.md:80]: This still gives the same "unable to find" error. I installed some necessary tools in the doc before installing Gazebo harmonic and it worked. We can add this:
`
sudo apt-get update
sudo apt-get install curl lsb-release gnupg
`

Comment (luca-byte) [reply README.md:80]: Yes, I added them in the development dependencies, but they can be moved here

**Review by Yunus-Selim — APPROVED** [review]
LGTM

**Review by Yusuf-Serdar — APPROVED** [review]
I've successfully run `colcon test` on my test code, and it works very well. The style changes are fine. LGTM!

**Review by lorixhack — APPROVED** [review]
It's fine for me, LGTM!

**Pull request merged**